### PR TITLE
Add base code for product compare feature

### PR DESCRIPTION
**Pull request by jose-correia**
## Notes
This PR does not implement the entire feature, but its the first iteration in order to not increase the LOC and make reviewing easier.

### TODO next

- [x] Fix the close compare buttons (X) which are not working
- [ ] Show modal with data when compare is clicked
- [ ] Styling polishing

@mhaack I added a basic compare button to each product item (via javascript). This is just for testing and will be removed when you finish your changes

Test URLs:
- Before: https://main--moleculardevices--hlxsites.hlx.live/products/cellular-imaging-systems
- After: https://400-compare-feature-1--moleculardevices--hlxsites.hlx.live/products/cellular-imaging-systems

### CODE REVIEW

File: blocks/product-carousel/product-carousel.js [hunk 1]
```diff
@@ -21,9 +23,9 @@ export default async function decorate(block) {
     products,
     {
       defaultStyling: true,
-      navButtons: window.matchMedia('only screen and (max-width: 1200px)').matches || products.length > 3,
+      navButtons: window.matchMedia('only screen and (max-width: 767px)').matches || products.length > 3,
       dotButtons: false,
-      infiniteScroll: products.length > 3,
```
Comment (mhaack) [inline blocks/product-carousel/product-carousel.js:26]: This must be reverted, it breaks the carousel behavior with <3 products. Like here: https://400-compare-feature-1--moleculardevices--hlxsites.hlx.live/products/microplate-readers/washers there should only be one "MultiWash+ Microplate Washer" in the list.

**Comment by lydiapuric** [comment]
@jose-correia , @duynguyen In our case, all products can be compared. This is not like this on the original page, see https://www.moleculardevices.com/products/cellular-imaging-systems

File: blocks/card/card.css [hunk 2]
```diff
@@ -91,6 +90,32 @@ main .card-caption .button.primary {
...
+.product-compare main .card-caption .c2a .compare-button {
+  display: flex;
+}
+
+main .card-caption .c2a .compare-checkbox {
+  width: 18px;
+  height: 18px;
+  margin-left: 10px;
+  vertical-align: middle;
+  border: 2px solid #adb3b7;
```
Comment (lydiapuric) [inline blocks/card/card.css:110]: same, use variable

File: blocks/card/card.css [hunk 3]
```diff
@@ -91,6 +90,32 @@ main .card-caption .button.primary {
...
+  cursor: pointer;
+}
+
+main .card-caption .c2a .compare-checkbox.selected {
+  background: #6eb43f;
```
Comment (lydiapuric) [inline blocks/card/card.css:115]: introduce variable for this color too

File: blocks/card/card.js [hunk 4]
```diff
@@ -1,11 +1,59 @@
...
+
+  // get or create compare banner
+  const compareBannerInterface = await createCompareBannerInterface({
+    currentCompareItemsCount: selectedItemPaths.length,
+    maxCompareItemsCount: MAX_COMPARE_ITEMS,
+  });
+
+  compareBannerInterface.getOrRenderBanner();
+
+  if (selectedItemPaths.includes(clickedItemPath)) {
```
Comment (lydiapuric) [inline blocks/card/card.js:34]: The readability could be improved here of the code by breaking up the if else statements and instead thinking in actions like, ADD and DELETE.

**Review by mhaack — APPROVED** [review]

**Pull request merged**